packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -339,6 +339,16 @@ class DevFS {
   Uri _baseUri;
   Uri get baseUri => _baseUri;
 
+  Uri deviceUriToHostUri(Uri deviceUri) {
```

Comment:
Wouldn't you want to use the deviceUri.path member (or some combination of members) here instead of converting it to a string and comparing the prefix?  Seems like you're throwing out the point of using a URI, which is that it gets parsed into its constituent parts so that things like hostname resolution and ports can be abstracted out.

Reply:
No, I'm looking for a complete prefix match. If the deviceUri has a prefix match only looking at paths, but differs in any other property, I want to return in as is.